tests/PHPStan/Analyser/TypeSpecifierTest.php

```diff
@@ -1283,7 +1283,7 @@ public function dataCondition(): iterable
 					),
 				),
 				[
-					'$foo' => 'non-empty-string',
+					'$foo' => "string & ~''",
```

Comment:
Isn't this change weird ?

Shouldn't `string & ~''` be considered as non-empty-string ?

Reply:
I think its the same thing, but just a difference how the type is printed